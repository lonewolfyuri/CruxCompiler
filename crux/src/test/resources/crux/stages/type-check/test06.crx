// test addition

func addInts(a:int, b:int) : int {
  return a + b;
}

func addBools(a:bool, b:bool) : bool {
  return a + b;
}

func addIntBool(a:int, b:bool) : void {
  return a + b;
}

func main() : void {
  ::printInt(7);
}

func addWithError() : void {
  var a : int;
  let a = 1 + 2 + 3 + false + 4 + 5 + 6;
}
