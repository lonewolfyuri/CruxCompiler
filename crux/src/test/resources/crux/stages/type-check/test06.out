Error type-checking file.
TypeError(8,11)[cannot add bool with bool]
TypeError(8,2)[Function addBools returns bool not ErrorType(cannot add bool with bool).]
TypeError(12,11)[cannot add int with bool]
TypeError(12,2)[Function addIntBool returns void not ErrorType(cannot add int with bool).]
TypeError(21,20)[cannot add int with bool]
TypeError(21,28)[cannot add ErrorType(cannot add int with bool) with int]
TypeError(21,32)[cannot add ErrorType(cannot add ErrorType(cannot add int with bool) with int) with int]
TypeError(21,36)[cannot add ErrorType(cannot add ErrorType(cannot add ErrorType(cannot add int with bool) with int) with int) with int]
TypeError(21,2)[cannot assign ErrorType(cannot add ErrorType(cannot add ErrorType(cannot add ErrorType(cannot add int with bool) with int) with int) with int) to Address(int)]
